src/libraries/System.Private.CoreLib/src/System/StringComparer.cs

```diff
@@ -62,6 +62,68 @@ public static StringComparer Create(CultureInfo culture, CompareOptions options)
             return new CultureAwareComparer(culture, options);
         }
 
+        public static bool IsWellKnownOrdinalComparer(IEqualityComparer<string?>? comparer, out bool ignoreCase)
+        {
+            if (comparer is IInternalStringEqualityComparer internalStringComparer)
+            {
+                comparer = internalStringComparer.GetUnderlyingEqualityComparer(); // unwrap if necessary
+            }
+
+            if (comparer is StringComparer stringComparer)
+            {
+                return stringComparer.IsWellKnownOrdinalComparerCore(out ignoreCase);
+            }
+            else if (comparer is GenericEqualityComparer<string>)
+            {
+                // special-case EqualityComparer<string>.Default, which is Ordinal-equivalent
+                ignoreCase = false;
+                return true;
+            }
+            else
+            {
+                // unknown comparer
+                ignoreCase = default;
+                return false;
+            }
+        }
+
+        // don't allow callers outside this hierarchy, and only allow overrides from within this assembly
```

Comment:
Nit: This comment seems unnecessary, since it's just spelling out the semantics of private protected virtual